core/src/main/java/io/stargate/core/activator/BaseActivator.java

```diff
@@ -63,19 +91,29 @@ public BaseActivator(String activatorName) {
   public synchronized void start(BundleContext context) throws InvalidSyntaxException {
     logger.info("Starting {} ...", activatorName);
     this.context = context;
-    if (dependencies().isEmpty() && lazyDependencies().isEmpty()) {
+
+    String filter = constructDependenciesFilter();
+
+    if (dependencies.isEmpty() && lazyDependencies().isEmpty()) {
       startServiceInternal();
     } else {
-      tracker = new Tracker(context, context.createFilter(constructDependenciesFilter()));
+      tracker = new Tracker(context, context.createFilter(filter));
       tracker.open();
     }
   }
 
   String constructDependenciesFilter() {
+    List<ServicePointer<?>> dep = dependencies();
+    if (healthCheck != null) {
+      dep = new ArrayList<>(dep);
```

Comment:
is this copy needed because you are taking into account the fact that `dependencies()` may return an immutable list?

Reply:
yes, some services return immutable dependencies lists.